src/Compilers/CSharp/Portable/Binder/Binder_Await.cs

```diff
@@ -209,17 +209,21 @@ private bool ReportBadAwaitContext(SyntaxNode node, DiagnosticBag diagnostics)
         /// Finds and validates the required members of an awaitable expression, as described in spec 7.7.7.1.
         /// </summary>
         /// <returns>True if the expression is awaitable; false otherwise.</returns>
-        private bool GetAwaitableExpressionInfo(
+        internal bool GetAwaitableExpressionInfo(
             BoundExpression expression,
             out MethodSymbol getAwaiter,
             out PropertySymbol isCompleted,
             out MethodSymbol getResult,
+            out BoundExpression getAwaiterGetResultCall,
             SyntaxNode node,
-            DiagnosticBag diagnostics)
+            DiagnosticBag diagnostics,
+            bool needsGetIsCompletedProperty = true,
+            bool needsAwaiterImplementsINotifyCompletion = true)
```

Comment:
>needsAwaiterImplementsINotifyCompletion [](start = 17, length = 39)

I don't think we should have these last two parameters, the requirements are part of the spec should be met for our case too. Even if we don't use them.